awx/main/tasks.py

```diff
@@ -906,6 +906,23 @@ def build_execution_environment_params(self, instance):
             "container_options": ['--user=root'],
         }
 
+        if instance.execution_environment.credential:
+            cred = instance.execution_environment.credential
+            if cred.has_inputs(field_names=('host', 'username', 'password')):
+                path = os.path.split(private_data_dir)[0]
+                with open(path + '/auth.json', 'w') as authfile:
+                    host = cred.get_input('host')
+                    username = cred.get_input('username')
+                    password = cred.get_input('password')
+                    token = "{}:{}".format(username, password)
+                    auth_data = {'auths': {host: {'auth': b64encode(token.encode('ascii')).decode()}}}
```

Comment:
Does this work if the password/token contains unicode characters, e.g., `Iñtërnâtiônàlizætiøn` ?

Reply:
the pass is defined by quay, so I haven't explicitly tested that, I definitely can though.